tests/engine_tests/test_engine.py

```diff
@@ -48,60 +42,135 @@ def _extract_test_cases(
     :param file_path: the path to the json data file
     :return: a list of tuples containing the environment, identity and api response
     """
-    with open(file_path, "r") as f:
-        test_data = EngineTestData.model_validate_json(f.read())
+    test_data = json.load(file_path.open("rb"))
```

Comment:
```suggestion
    test_data = json.loads(file_path.read_bytes())
```

Also: does it matter to read as binary?